captcha.module

```diff
@@ -144,8 +144,7 @@ function captcha_element_process(array $element, FormStateInterface $form_state,
   // This needs to be done even if the CAPTCHA will be ommitted later:
   // other untrusted users should not get a cached page when
   // the current untrusted user can skip the current CAPTCHA.
-  global $conf;
-  $conf['cache'] = FALSE;
+  \Drupal::service('page_cache_kill_switch')->trigger();
```

Comment:
Also, I think the comment there is confusing and incorrect (I don't think some anon users can be trusted and others not?

We also need a max-age 0 in the render array structure, so that it's not getting cached inside a block for example. We probably want to have an explicit tests for a user login block with the math or image plugin, visiting it multiple times should a) never get cached, and b) result in a different output.